### PR TITLE
Refactor transactions process loader

### PR DESCRIPTION
**Pull request by Pinkdesu**
- Removed lock screen loder
- Removed react-alert lib, created custom lib
- Changed upload program, upload code, send message  alert logic

### CODE REVIEW

File: website/frontend/src/context/alert/Provider.tsx [hunk 1]
```diff
@@ -1,26 +1,150 @@
...
+  const timers = useRef<AlertTimer>(new Map());
+  const [alerts, setAlerts] = useState<AlertInstance[]>([]);
+
+  const createTimer = useCallback((alertId: string, callback: (alertId: string) => void, timeout) => {
+    if (timeout > 0) {
+      const timerId = setTimeout(() => callback(alertId), timeout);
+
+      timers.current.set(alertId, timerId);
+    }
+  }, []);
```
Comment (nikitayutanov) [inline website/frontend/src/context/alert/Provider.tsx:27]:
```suggestion
  const createTimer = useCallback((alertId: string, timeout) => {
    if (timeout > 0) {
      const timerId = setTimeout(() => remove(alertId), timeout);

      timers.current.set(alertId, timerId);
    }
  }, []);
```
The only callback that is used right now is `remove`, and I highly doubt that something else could be useful

File: website/frontend/src/context/alert/Provider.tsx [hunk 2]
```diff
@@ -1,26 +1,150 @@
...
+
+      return id;
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [createTimer, remove]
+  );
 
-const AlertProvider = ({ children }: Props) => <Provider template={AlertTemplate} children={children} {...options} />;
+  const update = useCallback(
+    (id: string, content: ReactNode, options?: Omit<AlertOptions, 'castomId'>) => {
```
Comment (nikitayutanov) [inline website/frontend/src/context/alert/Provider.tsx:67]:
```suggestion
    (alertId: string, content: ReactNode, options?: Omit<AlertOptions, 'castomId'>) => {
```
Also, `AlertOptions` doesn't have `customId` key that you omit

File: website/frontend/src/context/alert/Provider.tsx [hunk 3]
```diff
@@ -1,26 +1,150 @@
...
+  const сreateTemplateAlert = useCallback(
+    (templateOptions: AlertOptions) => (content: ReactNode, options?: TemplateAlertOptions) =>
+      show(content, {
+        ...templateOptions,
+        ...options,
+      }),
+    [show]
+  );
+
+  const alertContext: AlertContainerFactory = useMemo(
```
Comment (nikitayutanov) [inline website/frontend/src/context/alert/Provider.tsx:105]:
```suggestion
  const alertContext = useMemo(
```

File: website/frontend/src/context/alert/Provider.tsx [hunk 4]
```diff
@@ -1,26 +1,150 @@
...
+      );
+
+      if (updatedAlert!) {
+        createTimer(updatedAlert.id, remove, updatedAlert.options.timeout);
+      }
+    },
+    [removeTimer, createTimer, remove]
+  );
+
+  const сreateTemplateAlert = useCallback(
```
Comment (nikitayutanov) [inline website/frontend/src/context/alert/Provider.tsx:96]:
```suggestion
  const getAlertTemplate = useCallback(
```
image
The first 'c' letter right now is written in cyrillic

File: website/frontend/src/components/Alert/Transition.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  entering: { opacity: 0 },
+  entered: { opacity: 1 },
+};
+
+type Props = {
+  children: ReactNode;
+};
+
+const Transition = ({ children, ...props }: Props) => {
+  const ref = useRef(null);
```
[the remark this answers is not in the document]
Comment (Pinkdesu) [reply website/frontend/src/components/Alert/Transition.tsx:21]: This ref is not only for 'div'. Transition use it

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**